### PR TITLE
V3 - FlatList signature, renamed components, performance improvements

### PR DESCRIPTION
**Pull request by saleel**
Exports two components FlatGrid and SectionGrid
FlatList like signature for FlatGrid
Common styles and calculation
Performance Improvements

### CODE REVIEW

**Comment by saleel** [comment]
@andersonaddo Please review this PR when you have time. I have refactored SectionGrid as well to use common functions.

File: SuperGridSectionList.js [hunk 1]
```diff
@@ -1,143 +1,142 @@
...
 
-  
   onLayout(e) {
-    const { staticDimension, onLayout } = this.props;
+    const { staticDimension, horizontal, onLayout } = this.props;
+    const { totalDimension } = this.state;
+
     if (!staticDimension) {
       const { width, height } = e.nativeEvent.layout || {};
+      const newTotalDimension = horizontal ? height : width;
```
[the remark this answers is not in the document]
Comment (andersonaddo) [reply SuperGridSectionList.js]: You did it for SuperGrid.js too. Was it intentional?

File: README.md [hunk 2]
```diff
@@ -193,19 +175,46 @@ const styles = StyleSheet.create({
...
-Please take a look on the follow example.
-
-```
+## SectionGrid Example
+```javascript
 import React, { Component } from 'react';
 import { StyleSheet, View, Text } from 'react-native';
-import GridView from 'react-native-super-grid';
+import { FlatGrid } from 'react-native-super-grid';
+
```
Comment (andersonaddo) [inline README.md]: This is meant to be an example for the SectionGrid, but it's FlatGrid code that's displayed here.

File: SuperGridSectionList.js [hunk 3]
```diff
@@ -1,143 +1,142 @@
...
-          <View key={`${item.key}_${i}`} style={itemContainerStyle}>
-            <View style={itemStyle}>
-              {this.props.renderItem({item: itemObject, index: i + (item.rowNumber * itemsPerRow), separators: separators, section: section })}
-            </View>
-          </View>
-        ))}
-      </View>
-    );
-  }
+    const groupedSections = sections.map(({ title, data }) => {
```
[the remark this answers is not in the document]
Comment (saleel) [reply SuperGridSectionList.js]: I did check those two issues. That was a fix for JSON.parse based cloning.
But I didn't see any particular reason we had to clone it. `chunkArray` do not modify the actual items. Do you remember why you had to clone it?
Map is working just fine. I did put some keys with function as values in the items and it worked fine.

File: README.md (hunk 2 above)
Comment (saleel) [reply README.md]: Yea. My bad. Fixing it shortly.

File: SuperGridSectionList.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (andersonaddo) [reply SuperGridSectionList.js]: My original code didn't use `.map`. It passed the original `sections` array directly into the `chunkArray` function, which ended up altering and condensing the `sections` array with every re-render.
I didn't realize I could just use `map`, that's why I cloned the whole array instead.

**Comment by andersonaddo** [comment]
I won't be able to test it out for a few days (laptop issues), but it looks okay from here. You can merge it now if you want, and I'll test it afterwards (especially since you've already tested it before).

**Comment by saleel** [comment]
Ok. I will merge it and release to npm. I have tested all scenarios and worked fine for me.

Thanks :+1:

**Pull request merged**